tests/core/useragent.c

```diff
@@ -0,0 +1,11 @@
+#include "clar_libgit2.h"
+#include "global.h"
+
+void test_core_useragent__get(void)
+{
+	const char *custom_name = "super duper git";
+
+	cl_assert_equal_p(NULL, git_libgit2__user_agent());
+	cl_git_pass(git_libgit2_opts(GIT_OPT_SET_USER_AGENT, custom_name));
+	cl_assert_equal_s(custom_name, git_libgit2__user_agent());
```

Comment:
Indeed it would be, but actually adding this to the testing is a large amount of work compared to the impact of this (and the impact of it breaking). We do want a way to test header fields in a more general way, and we will work on that to test higher-impact changes, but not for this largely cosmetic change.